### DISCUSSION TITLE
Query notifications sample

### DISCUSSION THREAD

**Discussion by the42** [microsoft/go-mssqldb#89] [Q&A]
https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/sql/detecting-changes-with-sqldependency provides an example how to register to mssql query notifications.

This driver supports Query notifications
https://github.com/microsoft/go-mssqldb#features

It is uncelar how this feature can be used in golang, a sample would be welcomed.

**Comment by shueybubbles** [comment]
It looks like the driver supports setting up the subscription but not consuming the changes, which are delivered by Service Broker. 
You'd add the subscription using `MssqlStmt.SetQueryNotification`
The feature was added 9 years ago https://github.com/microsoft/go-mssqldb/commit/8d4984e8baccbf5bfadd7f7e366fd61b7ccac38b
I don't see any Go code that issues the `RECEIVE` query to service broker to process the changes.

ADO.Net has a process-wide listener implemented here https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDependencyListener.cs

I'd say the Go driver implementation is far from complete.

**Comment by the42** [comment]
Thank you for you analysis!

> I'd say the Go driver implementation is far from complete.

Depends on the expected convenience level. Seen from your mentioned example of https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDependencyListener.cs and the boilerplate the .net class takes care of  - yes.

Given that consuming shouldn't be more than `RECEIVE` as pointed out in https://stackoverflow.com/questions/2158839/notify-a-c-application-when-a-change-in-a-sql-server-table-is-made - no.

But alas no public golang examples.

**Comment by kedzoo** [comment]
any update on how to use it?

**Comment by shueybubbles** [comment]
Just search for "RECEIVE TOP language:go" on github

This one pops up
https://github.com/yaxinr/amqp-mssql-listener
